etna/datasets/tsdataset.py

```diff
@@ -980,6 +980,63 @@ def has_hierarchy(self) -> bool:
         """Check whether dataset has hierarchical structure."""
         return self.hierarchical_structure is not None
 
+    def get_level_dataset(self, target_level: str) -> "TSDataset":
+        """Generate new TSDataset on target level.
+
+        Parameters
+        ----------
+        target_level:
+            target level name
+
+        Returns
+        -------
+        TSDataset
+            generated dataset
+        """
+        if not self.has_hierarchy():
+            raise ValueError("Method could be applied only to instances with a hierarchy!")
+
+        # ensure all necessary fields are initialized
+        assert self.hierarchical_structure is not None
+        assert self.current_df_level is not None
+
+        if target_level not in self.hierarchical_structure._level_to_index:
+            raise ValueError("Provided level name is not part of the hierarchy!")
+
+        current_level_index = self.hierarchical_structure._level_to_index[self.current_df_level]
+        target_level_index = self.hierarchical_structure._level_to_index[target_level]
+
+        if target_level_index > current_level_index:
+            raise ValueError("Target level should be higher in the hierarchy than the current level of dataframe!")
+
+        if target_level_index < current_level_index:
+            summing_matrix = self.hierarchical_structure.get_summing_matrix(
+                target_level=target_level, source_level=self.current_df_level
+            )
+
+            current_level_segments = self.hierarchical_structure.get_level_segments(level_name=self.current_df_level)
```

Comment:
Place this instead of the "if" checking the existence of `target_level` to make sure that the levels are correct